test

```diff
@@ -57,26 +57,61 @@ fi
 split=(${TEST// / })
 TEST=${split[@]/#/${REPO_PATH}/}
 
+# PKGS_DELIM contains all the core etcd pkgs delimited by ',' which will be profiled for code coverage.
+# Integration and e2d tests will generate code coverage for those pkgs 
+export PKGS_DELIM=$(echo $TEST | sed 's/ /,/g')
+
 # determine whether target supports race detection
 if [ "$GOARCH" == "amd64" ]; then
 	RACE="--race"
 fi
 
+export COVEROUT="coverage"
+
+# create coverage folder if it doesn't exist. if it exist, then wipe out its content for new coverage results.
+if ! [ -d "$COVEROUT" ]; then
+	mkdir "$COVEROUT"
+else
+	rm -f "$COVEROUT"/*
+fi
+
 function unit_pass {
 	echo "Running unit tests..."
 	# only -run=Test so examples can run in integration tests
-	go test -timeout 3m ${COVER} ${RACE} -cpu 1,2,4 -run=Test $@ ${TEST}
```

Comment:
The coverage analysis shouldn't be baked into the tests by default; it's a bad performance hit. Try having a `cov_pass` function instead that runs all the tests with coverage enabled?